near-sdk/src/collections/heap_map.rs

```diff
@@ -0,0 +1,183 @@
+use std::marker::PhantomData;
+use borsh::{BorshDeserialize, BorshSerialize};
+
+use crate::collections::{append, serialize, deserialize, Heap, append_slice};
+use crate::env;
+
+/// HeapMap allows iterating over keys and entries based on natural key ordering.
+///
+/// Runtime complexity (worst case):
+///   - `contains_key`: O(1)
+///   - `get`: O(1)
+///   - `insert`: O(log(N))
+///   - `remove`: O(log(N))
+///   - `keys` (iterator): O(Nlog(N))
+///   - `entries` (iterator): O(Nlog(N))
+#[derive(BorshSerialize, BorshDeserialize)]
+pub struct HeapMap<K, V> {
+    key_prefix: Vec<u8>,
+    keys: Heap<K>,
+    _values: PhantomData<V>,
+}
+
+impl<K, V> HeapMap<K, V>
+    where
+        K: Ord + BorshSerialize + BorshDeserialize,
+        V: BorshSerialize + BorshDeserialize,
+{
+    pub fn new(id: Vec<u8>) -> Self {
```

Comment:
Please add method that exposes the largest key (the top element from the heap).

Reply:
done